ui/src/layout/Layout.js

```diff
@@ -1,19 +1,31 @@
 import React, { useCallback } from 'react'
 import { useDispatch, useSelector } from 'react-redux'
 import { Layout, toggleSidebar } from 'react-admin'
+import useMediaQuery from '@material-ui/core/useMediaQuery'
 import { makeStyles } from '@material-ui/core/styles'
 import { HotKeys } from 'react-hotkeys'
 import Menu from './Menu'
 import AppBar from './AppBar'
 import Notification from './Notification'
 import themes from '../themes'
+import { AUTO } from '../consts'
 
 const useStyles = makeStyles({
   root: { paddingBottom: (props) => (props.addPadding ? '80px' : 0) },
 })
 
+export const useCurrentTheme = () => {
+  const prefersLightMode = useMediaQuery('(prefers-color-scheme: light)')
+  return useSelector((state) => {
+    if (state.theme === AUTO) {
+      return prefersLightMode ? themes.LightTheme : themes.DarkTheme
+    }
+    return themes[state.theme] || themes.DarkTheme
+  })
+}
+
```

Comment:
Please extract the hook to its own file, and move it to the `themes` folder: `ui/themes/useCurrentTheme.js`

Reply:
Done!